src/app.js

```diff
@@ -1,12 +1,47 @@
+import { getCoffeeShops } from './api.js';
+import {
+    calculateDistance,
+    getLowestThreeIndices,
+    validatePosition,
+} from './utils.js';
+
 /**
  * @param {Object} position
  * @param {Number} position.x
  * @param {Number} position.y
- * 
+ *
  * @returns {Array<position>}
  */
-export function getNearestShops(position) {
-  // code
-  
-  return [];
+export async function getNearestShops(position) {
+    try {
+        const validatedPosition = validatePosition(position);
+        const coffeeShops = await getCoffeeShops();
+
+        if (!Array.isArray(coffeeShops)) {
+            throw new Error('Invalid data returned by API.');
+        }
+
+        const distances = coffeeShops.map(shop => {
+            const distance = calculateDistance(
+                { x: shop.x, y: shop.y },
+                validatedPosition,
+            );
+
+            return {
+                name: shop.name,
+                distance: distance,
+            };
+        });
```

Comment:
Nothing wrong with the way it is, you could just write this a bit simpler

```suggestion
        const distances = coffeeShops.map(shop => ({
            name: shop.name
            distance: calculateDistance(
                { x: shop.x, y: shop.y },
                validatedPosition,
            )
        });
```